txp/Pos/Txp/Logic/Local.hs

```diff
@@ -62,35 +72,52 @@ txProcessTransaction itw@(txId, txAux) = do
                    toList $
                    NE.zipWith (liftM2 (,) . Just) _txInputs resolvedOuts
     pRes <- modifyTxpLocalData $
-            processTxDo resolved toilEnv tipDB itw
+            processTxDo resolved bootHolders toilEnv tipDB itw bootEra
     case pRes of
         Left er -> do
             logDebug $ sformat ("Transaction processing failed: "%build) txId
             throwError er
         Right _   ->
             logDebug (sformat ("Transaction is processed successfully: "%build) txId)
   where
+    notBootRelated bootHolders =
+        let txDistr = getTxDistribution $ taDistribution txAux
+            inBoot s = s `HS.member` genesisBootProdStakeholders
+            mentioned pool addr = addr `elem` pool
+            bad :: [(StakeholderId, Coin)] -> Bool
+            bad (map fst -> txOutDistr) =
+                -- Has unrelated address
+                any (not . inBoot) txOutDistr ||
+                -- Not all genesis boot addrs are mentioned
+                any (not . mentioned txOutDistr) (HS.toList bootHolders)
```

Comment:
You open daedalus. You send 3 coins. You see some weird stuff instead of understandable error message. I think it's bad.